library/general/src/lib/ui/installation/layout.rb

```diff
@@ -0,0 +1,166 @@
+    #   layout2 = UI::Installation::Layout.with_title_on_left
+    #   layout3 = UI::Installation::Layout.with_title_on_top
+    #
+    #   layout3.show_banner
+    #
+    #   layout1.open_wizard do
+    #     # ...
+    #   end
+    class Layout
+      Yast.import "UI"
+      Yast.import "Wizard"
+
+      class << self
+        # Creates a new layout with a left sidebar to place the installation steps
+        #
+        # @return [Layout]
+        def with_steps
+          config = LayoutConfig.new.tap { |c| c.mode = LayoutConfig::Mode::STEPS }
+
+          new(config)
+        end
+
+        # Creates a new layout with without a sidebar but with the title of the dialogs on the left
+        #
+        # @return Layout]
+        def with_title_on_left
+          config = LayoutConfig.new.tap { |c| c.mode = LayoutConfig::Mode::TITLE_ON_LEFT }
+
+          new(config)
+        end
+
+        # Creates a new layout with without a sidebar but with the title of the dialogs on top
+        #
+        # @return [Layout]
+        def with_title_on_top
+          config = LayoutConfig.new.tap { |c| c.mode = LayoutConfig::Mode::TITLE_ON_TOP }
+
+          new(config)
+        end
+
+        # Creates a new layout according to the product features
+        #
+        # @see LayoutConfig.from_product_features
+        #
+        # @return [Layout]
+        def from_product_features
+          config = LayoutConfig.from_product_features
+
+          new(config)
+        end
+      end
+
+      # Configures the layout to show a banner
+      def show_banner
+        config.banner = true
+      end
+
+      # Configures the layout to not show a banner
+      def hide_banner
+        config.banner = false
+      end
+
+      # Whether the layout includes a banner
+      #
+      # @return [Boolean]
+      def banner?
+        config.banner
+      end
+
+      # Whether the layout has a left sidebar to place the installation steps
+      #
+      # @return [Boolean]
+      def with_steps?
+        config.mode == LayoutConfig::Mode::STEPS
+      end
+
+      # Whether the layout has the title of the dialogs on the left
+      #
+      # @return [Boolean]
+      def with_title_on_left?
+        config.mode == LayoutConfig::Mode::TITLE_ON_LEFT
+      end
+
+      # Whether the layout has the title of the dialogs on the top
+      #
+      # @return [Boolean]
+      def with_title_on_top?
+        config.mode == LayoutConfig::Mode::TITLE_ON_TOP
+      end
+
+      # Opens a new wizard according to the layout configuration
+      #
+      # @yield Code to run after opening the wizard. The wizard is automatically closed.
+      def open_wizard(&block)
+        Yast::UI.SetProductLogo(banner?)
+
+        case config.mode
+        when LayoutConfig::Mode::STEPS
+          Yast::Wizard.OpenNextBackStepsDialog
+        when LayoutConfig::Mode::TITLE_ON_LEFT
+          Yast::Wizard.OpenLeftTitleNextBackDialog
+        else
+          Yast::Wizard.OpenNextBackDialog
```

Comment:
To be honest I do not like much all these comparison to constant. It is a too much for me and here is also hidden default, which is not mentioned anywhere else. So what I propose is to create also class for mode that has those constants and it will return object with methods like `steps?`, `title_left?`, `title_top?` or wizard_method which can be used e.g. here as 
`Yast::Wizard.public_send(config.mode.wizard_method))`

Reply:
ah, lower I see default, so here that else does not make sense at all for me and is just confusing. I would at least use when for title on top and else which raise for invalid value.